apps/news/news.arc

```diff
@@ -1008,14 +1008,17 @@
 ; since only uparrow is shown; could straight memoize
 
 (def votelink (i user whence dir)
-  (tag (a id      (if user (string dir '_ i!id))
-          onclick (if user "return vote(this)")
```

Comment:
Good patch!

Only nitpick is that the vote JS function is triggered regardless of whether a user is logged in, where they should really be redirected to the login page.

Reply:
True... that could probably be fixed by posting an actual JSON request to the vote url instead of the weird image thing it's doing now) and redirecting based on a response code.

Reply:
> and redirecting based on a response code.

I'd probably just indicate login status in a votelink tag attribute as that would mean one less roundtrip. But your solutions sounds good, too, and way less hacky than what's going on now

On HN they [still](https://news.ycombinator.com/hn.js) use that weird image hack, even though they have an `ajax` function defined. Not sure I get it, but there seems to be this thing going on in Arc of misusing images as much as possible, such as doing all the spacing and indentation by stretching a 1x1 transparent GIF, rather than, say, CSS.